charlib/liberty/liberty.py

```
@@ -72,15 +86,14 @@ def add_group(self, group, group_id=''):
         except KeyError:
             pass
         finally:
-            self.groups[hash(group)] = group
+            self.groups[group.unique_key] = group
 
-    def group(self, name, identifier=''):
-        """Look up a sub-group by name and id"""
+    def group(self, *key):
```

Ruff (0.15.20)

[warning] Missing type annotation for `*key`

(ANN002)